plonky3/src/prover/mod.rs

```diff
@@ -97,13 +98,68 @@ mod tests {
         assert!(proof.is_ok());
     }
 
+    fn run_test_goldilocks_publics(
+        pil: &str,
+        malicious_publics: Vec<GoldilocksField>,
+    ) -> Result<(), String> {
+        let mut pipeline = Pipeline::<GoldilocksField>::default().from_pil_string(pil.to_string());
+
+        let pil = pipeline.compute_optimized_pil().unwrap();
+        let witness_callback = pipeline.witgen_callback().unwrap();
+        let witness = pipeline.compute_witness().unwrap();
+
+        let prover = Plonky3Prover::new(&pil);
+        let proof = prover.prove(&witness, witness_callback).unwrap();
+
+        prover.verify(&proof, &[malicious_publics])
+    }
+
     #[test]
-    #[should_panic = "not implemented"]
+    // #[should_panic = "not implemented"]
     fn publics() {
         let content = "namespace Global(8); pol witness x; x * (x - 1) = 0; public out = x(7);";
         run_test_goldilocks(content);
     }
 
+    #[test]
+    fn public_inputs() {
+        let content = r#"
+        namespace Add(8);
+            col witness x;
+            col witness y;
+            col witness z;
+            y - 1 = 0;
+            x = 0;
+            x + y = z;
+
+            public outz = z(7);
+        "#;
+        let publics = vec![GoldilocksField::from(1)];
```

Comment:
We can use `run_test_goldilocks` for the honest case, the prover already calls the verifier internally